sdk/servicebus/Azure.Messaging.ServiceBus/api/Azure.Messaging.ServiceBus.netstandard2.0.cs

```diff
@@ -240,6 +240,7 @@ public partial class ServiceBusReceivedMessage
         internal ServiceBusReceivedMessage() { }
         public System.Collections.Generic.IReadOnlyDictionary<string, object> ApplicationProperties { get { throw null; } }
         public System.BinaryData Body { get { throw null; } }
+        public Azure.Core.Amqp.AmqpMessageBodyType BodyType { get { throw null; } }
```

Comment:
Isn't this a slight violation of the current design? It seems that the received message is a higher level type that kind of abstracts from the underlying AMQP types. By adding this property the AMQP information is exposed on the higher level type. So if my design assumptions are true wouldn't it be better to have a well defined enum that maps the underlying type or a stringified representation of the body type instead of exposing the AMQP type?

Reply:
I'm curious what the scenario is where this would be needed at all?   If we accept the majority case as publishing and consuming via the Service Bus SDK message abstractions, I wonder if this property would be more confusing to most customers.   

I'd think that only those who know that they're in interop or advanced scenarios and already have awareness of the AMQP-level implications would benefit from this.  I would expect developers with those needs to be working with the `AmqpAnnotatedMessage` anyway.

Reply:
> Isn't this a slight violation of the current design? It seems that the received message is a higher level type that kind of abstracts from the underlying AMQP types. By adding this property the AMQP information is exposed on the higher level type. So if my design assumptions are true wouldn't it be better to have a well defined enum that maps the underlying type or a stringified representation of the body type instead of exposing the AMQP type?

You are right that it is somewhat leaking the AMQP details, but the goal here is to make it more convenient for consumers of an entity that may contain non-Data body messages. It's possible that users may not know whether their entity will ever contain such messages and they can use this property to ensure that their call to the Body getter will not throw.
Can you expand on what you mean by a well defined enum that maps the type? That is what this type is.

> I'd think that only those who know that they're in interop or advanced scenarios and already have awareness of the AMQP-level implications would benefit from this. I would expect developers with those needs to be working with the AmqpAnnotatedMessage anyway.

Yeah I had this thought as well, but the counterargument is that we want to provide a convenient way to inspect this up front without hitting exceptions. There may be instances where a consumer will just ignore non-Data messages or forward them somewhere else rather than wanting to dig into the message body.

Reply:
Something else that we can consider that would not violate the hierarchical design (or at least less so), would be to have a bool property like `IsDataBody`.

After thinking about this more, and considering your input, I'd prefer to close this until we get more feedback indicating that something like this property is important.